manifests/export/pem_cert.pp

```diff
@@ -4,30 +4,54 @@
 #
 # == Parameters
 #   [*pfx_cert*]  - PFX certificate/key container
+#   [*der_cert*]  - DER certificate
 #   [*pem_cert*]  - PEM/x509 certificate
 #   [*in_pass*]   - PFX password
 #
-define openssl::export::pem_cert(
-  Stdlib::Absolutepath      $pfx_cert,
-  Stdlib::Absolutepath      $pem_cert = $title,
-  Enum['present', 'absent'] $ensure   = present,
-  Optional[String]          $in_pass  = undef,
+define openssl::export::pem_cert (
+  Optional[Stdlib::Absolutepath]  $pfx_cert = undef,
+  Optional[Stdlib::Absolutepath]  $der_cert = undef,
+  Stdlib::Absolutepath            $pem_cert = $title,
+  Enum['present', 'absent']       $ensure   = present,
+  Optional[String]                $in_pass  = undef,
+
 ) {
-  if $ensure == 'present' {
-    $passin_opt = $in_pass ? {
-      undef   => '',
-      default => "-passin pass:'${in_pass}'",
-    }
+  #local variables
+
+  # If ensure = present and  der_cert and $pfx_cert as being specified, then throw error
+  if (($ensure == present) and ($der_cert == undef ) and ($pfx_cert == undef )) {
+    fail('Parameter Error: either pfx_cert or der_cert must be specified')
+  }
+
+  if (($der_cert != undef ) and ($pfx_cert != undef )) {
```

Comment:
Since these variables are strings, they can only evaluate to false if they're `undef`

```suggestion
  if $der_cert and $pfx_cert {
```